x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/sort/DoubleBucketedSortTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data.sort;
+
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.DoubleBlock;
+import org.elasticsearch.compute.data.ElementType;
+import org.elasticsearch.compute.data.IntVector;
+import org.elasticsearch.search.sort.SortOrder;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class DoubleBucketedSortTests extends BucketedSortTestCase<DoubleBucketedSort> {
+    @Override
+    protected DoubleBucketedSort build(SortOrder sortOrder, int bucketSize) {
+        return new DoubleBucketedSort(bigArrays(), sortOrder, bucketSize);
+    }
+
+    @Override
+    protected Object expectedValue(double v) {
+        return v;
+    }
+
+    @Override
+    protected double randomValue() {
+        // 2L^50 fits in the mantisa of a double which the test sort of needs.
+        return randomLongBetween(-2L ^ 50, 2L ^ 50);
```

Comment:
I like this one:
```
if (randomBoolean()) {
  return randomDouble();
} else {
  return Long.doubleFromLongBitsOrWhateverTheMethodIsCalled(randomLong());
}
```
That way you can get a truly wild smattering of `double` values.

Reply:
Maybe we'd have to intentionally skip `NaN` and `Inf` if it comes out of this - ESQL doesn't emit those values in `DoubleBlock`. We make a warning and `null` instead.

Reply:
Oh, I copypasted it, and forgot to replace that long logic to double logic. Changed with randomDoubleBetween(min, max).

For Long and Int however, as it works by casting things, we would have to redo those tests. As it's a copy of the original BucketedSort tests, I'm leaving it that way for now; I don't think it's worth it to rewrite them now (?)
Maybe if we add BytesRef support later it would be a nice time to revisit this

Reply:
:+1: